### ISSUE TITLE
nitrogen6x: Can't write on U-Boot environment

### ISSUE THREAD

**Issue by Ossanes** [Freescale/meta-freescale-3rdparty#211]
Image: core-image-base

I'm using fw_printenv (from u-boot-fw-utils) to read the environment.

fw_env.config:
`/dev/mtd1   0x00000 0x2000  0x1000  2`

When I write any variable to the env using fw_setenv, I can't read it anymore.

```
root@nitrogen6x:/# fw_printenv
Cannot read environment, using default
Cannot read default environment from file
```

@gibsson

**Comment by gibsson** [comment]
@chrisdimich could you look into that, is it ok on our honister image? Thanks

**Comment by gibsson** [comment]
Yes, this is a bug in the MTD driver of the kernel which is broken since 5.4 (up to 5.15). When you write from userspace only the first byte is written to, so CRC is wrong. We hope to have a solution soon. We'll update once it's fixed.
